scanner/src/main/kotlin/scanners/fossid/FossIdNamingProvider.kt

```diff
@@ -47,26 +49,33 @@ class FossIdNamingProvider(
     }
 
     fun createProjectCode(projectName: String): String = namingProjectPattern?.let {
-        replaceNamingConventionVariables(namingProjectPattern, projectName, namingConventionVariables)
+        val builtins = mapOf(
+            "#projectName" to projectName
+        )
+        replaceNamingConventionVariables(namingProjectPattern, builtins, namingConventionVariables)
     } ?: projectName
 
-    fun createScanCode(projectName: String): String {
-        val pattern = namingScanPattern ?: "${projectName}_#currentTimestamp"
-        return replaceNamingConventionVariables(pattern, projectName, namingConventionVariables)
+    fun createScanCode(projectName: String, deltaTag: FossId.DeltaTag? = null): String {
+        val defaultPattern = "${projectName}_#currentTimestamp${if (deltaTag == null) "" else "_#deltaTag"}"
```

Comment:
BTW, it just occurred to me: Why is `${projectName}` already inserted here? Shouldn't it be `#projectName` to let `replaceNamingConventionVariables()` do the replacement?

Reply:
Oh yes, well spotted.